src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -8826,12 +8826,20 @@ public String clusterFailover(ClusterFailoverOption failoverOption) {
   }
 
   @Override
+  @Deprecated
   public List<Object> clusterSlots() {
     checkIsInMultiOrPipeline();
     connection.sendCommand(CLUSTER, ClusterKeyword.SLOTS);
     return connection.getObjectMultiBulkReply();
   }
 
+  @Override
+  public List<Object> clusterShards() {
+    checkIsInMultiOrPipeline();
+    connection.sendCommand(CLUSTER, ClusterKeyword.SHARDS);
+    return connection.getObjectMultiBulkReply();
```

Comment:
We can use this opportunity to introduce a custom encoded object, e.g. create new ClusterShardInfo and return List\<ClusterShardInfo\> here.

Reply:
@yangbodong22011 WDYT?

Reply:
> @yangbodong22011 WDYT?

Totally agree and this will also help us in the future when we **use cluster shards instead of cluster nodes**(maybe).

Reply:
@wormogig In this PR would be great!

Reply:
@sazzad16 @yangbodong22011 
I have created classes for command response CLUSTER SHARDS. Please review.